clang/include/clang/AST/DeclTemplate.h

```diff
@@ -1581,26 +1581,33 @@ class TemplateTemplateParmDecl final
       DefaultArgStorage<TemplateTemplateParmDecl, TemplateArgumentLoc *>;
   DefArgStorage DefaultArgument;
 
+  /// Whether this template template parameter was declaration with
+  /// the 'typename' keyword.
+  ///
+  /// If false, it was declared with the 'class' keyword.
+  bool Typename : 1;
```

Comment:
I like the bitfield nature here, but would prefer we do an 'unsigned' for each, and just 'treat' them as a bool later (it is a bit of a convention here).

Additionally, `Typename` should have an enum value that it gets compared to, rather than true meaning typename, false meaning class.

Reply:
Yup, i believe using bool is still causing issues for msvc

Reply:
Corentin's point is relevant here, you can't really use 'bool' in these (and should use the PREFERRED_TYPE or whatever attribute) thanks to some versions of MSVC we support.